stdlib/Sockets/src/Sockets.jl

```diff
@@ -600,6 +600,24 @@ function quickack(sock::Union{TCPServer, TCPSocket}, enable::Bool)
     nothing
 end
 
+"""
+    keepalive(sock::Union{TCPServer, TCPSocket}, enable::Bool, delay::Int)
+
+Enable / disable TCP keep-alive. `delay` is the initial delay in seconds, ignored when `enable`` is false.
+After delay has been reached, 10 successive probes, each spaced 1 second from the previous one, will still happen.
+If the connection is still lost at the end of this procedure, then the handle is destroyed with a UV_ETIMEDOUT error passed to the corresponding callback.
+
+!!! compat "Julia 1.11"
+    This function requires Julia 1.11 or later.
+"""
+function keepalive(sock::Union{TCPServer, TCPSocket}, enable::Bool, delay::Int)
+    iolock_begin()
+    check_open(sock)
```

Comment:
`check_open` can throw, so we need to wrap it in a try-finally for the `iolock_end()`

Reply:
iolock is implicitly reset by throw without additional handling